src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlSerializationWriterILGen.cs

```diff
@@ -2416,15 +2416,27 @@ internal void ILGenForCreateInstance(CodeGenerator ilg, Type type, Type cast, bo
                 }
             }
 
-            MethodInfo Activator_CreateInstance = typeof(Activator).GetMethod(
-                  "CreateInstance",
-                  CodeGenerator.StaticBindingFlags,
-                  new Type[] { typeof(Type) }
-                  );
-            ilg.Ldc(type);
-            ilg.Call(Activator_CreateInstance);
-            if (cast != null)
-                ilg.ConvertValue(Activator_CreateInstance.ReturnType, cast);
+            Type formatterServiceType = typeof(string).GetTypeInfo().Assembly.GetType("System.Runtime.Serialization.FormatterServices");
+            MethodInfo getUninitializedObject = formatterServiceType != null
+                ? formatterServiceType.GetMethod("GetUninitializedObject", BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static)
+                : null;
+            if (getUninitializedObject != null)
```

Comment:
This seems like it's always going to prefer GetUninitializedObject if it exists.  Wouldn't we want to actually prefer using the ctor if it has an accessible one?